extras/rya.streams/query-manager/src/main/java/org/apache/rya/streams/querymanager/QueryManager.java

```diff
@@ -0,0 +1,241 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.rya.streams.querymanager;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import org.apache.rya.streams.api.entity.StreamsQuery;
+import org.apache.rya.streams.api.queries.ChangeLogEntry;
+import org.apache.rya.streams.api.queries.InMemoryQueryRepository;
+import org.apache.rya.streams.api.queries.QueryChange;
+import org.apache.rya.streams.api.queries.QueryChange.ChangeType;
+import org.apache.rya.streams.api.queries.QueryChangeLog;
+import org.apache.rya.streams.api.queries.QueryChangeLogListener;
+import org.apache.rya.streams.api.queries.QueryRepository;
+import org.apache.rya.streams.querymanager.QueryChangeLogSource.SourceListener;
+import org.apache.rya.streams.querymanager.QueryExecutor.QueryExecutorException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.common.util.concurrent.AbstractScheduledService.Scheduler;
+
+import edu.umd.cs.findbugs.annotations.DefaultAnnotation;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * A service for managing {@link StreamsQuery} running on a Rya Streams system.
+ * <p>
+ * Only one QueryManager needs to be running to manage any number of rya
+ * instances/rya streams instances.
+ */
+@DefaultAnnotation(NonNull.class)
+public class QueryManager extends AbstractIdleService {
+    private static final Logger LOG = LoggerFactory.getLogger(QueryManager.class);
+
+    private final String ryaInstanceName;
+    private final QueryExecutor queryExecutor;
+    private final Scheduler scheduler;
+    private final Set<QueryRepository> queryRepos;
+
+    private final Map<String, QueryChangeLogSource> sources;
+    private final Map<UUID, StreamsQuery> queriesCache;
+
+    /**
+     * Creates a new {@link QueryManager}.
+     *
+     * @param ryaInstanceName - The Rya Instance to connect to. (not null)
+     * @param queryExecutor - The {@link QueryExecutor} that starts/creates
+     *        queries when a CREATED query is found in the
+     *        {@link QueryChangeLog}. (not null)
+     * @param sources - The {@link QueryChangeLogSource}s the sources of
+     *        QueryChangeLogs. (not null)
+     * @param queryRepos - The set of {@link QueryRepository}s that will notify
+     *        this manager when {@link ChangeLogEntry}s occur. (not null)
+     * @param scheduler - The {@link Scheduler} to use throughout
+     */
+    public QueryManager(final String ryaInstanceName, final QueryExecutor queryExecutor,
+            final Map<String, QueryChangeLogSource> sources, final Set<QueryRepository> queryRepos,
```

Comment:
You shouldn't be passing query repositories into this constructor since all of them have to be discovered through the QueryChangeLogSources